docset/winserver2025-ps/smbshare/Grant-SmbClientAccessToServer.md

```diff
@@ -0,0 +1,285 @@
+external help file: SmbServerConfiguration.cdxml-help.xml
+Module Name: SmbClientAccessToServer
+ms.date: 02/22/2024
+online version: https://learn.microsoft.com/powershell/module/smbshare/grant-smbclientaccesstoserver?view=windowsserver2025-ps&wt.mc_id=ps-gethelp
+schema: 2.0.0
+title: Grant-SmbClientAccessToServer
+---
+
+# Grant-SmbClientAccessToServer
+
+## SYNOPSIS
+Grants SMB client access to a specified server.
+
+## SYNTAX
+
+```
+Grant-SmbClientAccessToServer [-Name] <String[]> -IdentifierType <IdentifierType>
+ -Identifier <String> [-Description <String>] [-Force] [-CimSession <CimSession[]>]
+ [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+```
+Grant-SmbClientAccessToServer -InputObject <CimInstance[]> -IdentifierType <IdentifierType>
+ -Identifier <String> [-Description <String>] [-Force] [-CimSession <CimSession[]>]
+ [-ThrottleLimit <Int32>] [-AsJob] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+The `Grant-SmbClientAccessToServer` cmdlet is used to allow SMB clients to access a specified
+server. When you run this cmdlet, it'll add a new rule to the server's firewall to allow incoming
+SMB traffic from the specified client IP address. If you need to revoke access to the server in the
+future, you can use the `Revoke-SmbClientAccessToServer` cmdlet.
+
+## EXAMPLES
+
+### Example 1: Grant SMB client access to a server
+
+```powershell
+$params = @{
+    Name = "Server01"
+    IdentifierType = "ISSUER"
+    Identifier = "CN=MyCertificateIssuer"
+    Description = "Allow access from client for LabSvr"
+    Force = $true
+}
+Grant-SmbClientAccessToServer @params
+```
+
+This example grants SMB client access to a server named "Server01" based on the client's
+certificate. The **Description** parameter is used to provide a description of the rule being added
+to the server's firewall. This command runs without prompting for confirmation.
+
+## PARAMETERS
+
+### -AsJob
+
+Runs the cmdlet as a background job. Use this parameter to run commands that take a long time to
+complete.
+
+```yaml
+Type: SwitchParameter
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -CimSession
+
+Runs the cmdlet in a remote session or on a remote computer. Enter a computer name or a session
+object, such as the output of a [New-CimSession](/powershell/module/cimcmdlets/new-cimsession) or
+[Get-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227966) cmdlet. The default is the
+current session on the local computer.
+
+```yaml
+Type: CimSession[]
+Parameter Sets: (All)
+Aliases: Session
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Confirm
+
+Prompts you for confirmation before running the cmdlet.
+
+```yaml
```

Comment:
**Confirm** and **WhatIf** parameters go at the end of the list before CommonParameters.